library/swift/test/EnvoyClientTests.swift

```diff
@@ -67,7 +67,7 @@ final class EnvoyClientTests: XCTestCase {
       method: .get, scheme: "https", authority: "www.envoyproxy.io", path: "/docs")
```

Comment:
do we want to protect at the core layer against `sendHeaders` calls that do not set an authority header?

Reply:
That's a good callout and something we've discussed before. Right now we don't check to see if this is a valid HTTP/2 request at this layer, but we certainly could. I think it's worth discussing and/or tracking in an issue, but somewhat orthogonal to this change (since we haven't done it before either).

Reply:
FWIW `authority` is a required property when initializing the `RequestBuilder`, so consumers would have to explicitly set an invalid/empty authority value here. Could be worth asserting or logging in the core layer as you mentioned, though